.github/workflows/upload.yml

```diff
@@ -0,0 +1,77 @@
+name: upload
+
+on:
+  workflow_call:
+    inputs:
+      RELEASE_NAME:
+        type: string
+      ENVIRONMENT:
+        type: string
+      LATEST:
+        type: boolean
+      NIGHTLY:
+        type: boolean
+    secrets:
+      AWS_REGION:
+        required: true
+      AWS_ACCESS_KEY_ID:
+        required: true
+      AWS_SECRET_ACCESS_KEY:
+        required: true
+      RELEASES_BUCKET:
+        required: true
+      HELM_BUCKET:
+        required: true
+      DOCKERHUB_USERNAME:
+        required: true
+      DOCKERHUB_PASSWORD:
+        required: true
+      TELEMETRY_WRITE_KEY:
+        required: true
+
+jobs:
+  upload-binaries:
+    runs-on: ubuntu-latest
+    environment: ${{ inputs.ENVIRONMENT }}
+    steps:
+      - uses: aws-actions/configure-aws-credentials@v1
```

Comment:
Ideally this uses YAML anchors so it doesn't need to be repeated in `upload-helm` but that's not support currently